crates/transport-http/src/hyper.rs

```diff
@@ -30,11 +30,22 @@ where
 
             let resp = this.client.request(req).await.map_err(TransportErrorKind::custom)?;
 
-            // unpack json from the response body
+            let status = resp.status();
+
+            // unpack data from the response body. We do this regardless of
+            // the status code, as we want to return the error in the body if
+            // there is one.
             let body = hyper::body::to_bytes(resp.into_body())
                 .await
                 .map_err(TransportErrorKind::custom)?;
 
+            if status != hyper::StatusCode::OK {
+                return Err(TransportErrorKind::custom_str(&format!(
+                    "HTTP error: {} with body: {:?}",
+                    status, body
```

Comment:
perhaps we get the respons as text if not Ok:

if !=Ok use body.text() which sanitizes non utf8 iirc

else we get the raw body

Reply:
oh I guess this is raw hyper not reqwest
I believe reqwest does utf8 decoding on `.text()`

we could convert it to utf8 lossy perhaps, not sure how debug formats here